src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php

```diff
@@ -132,7 +132,11 @@ private function sanitizeQuery($connectionName, $query)
                 }
                 if ($type instanceof Type) {
                     $query['types'][$j] = $type->getBindingType();
-                    $param = $type->convertToDatabaseValue($param, $this->registry->getConnection($connectionName)->getDatabasePlatform());
+                    try {
+                        $param = $type->convertToDatabaseValue($param, $this->registry->getConnection($connectionName)->getDatabasePlatform());
+                    } catch (\Throwable $e) {
```

Comment:
This should be more specific, catching any throwable silently should be avoided.

Reply:
Thinking about it: a `ConversionException` can be caught instead of `\Throwable`, but others error (such as a `\TypeError`) can be thrown while processing query params.
In this case the error will be also thrown by DBAL directly and this is *only* a profiler data collector and should not throw an error to the user.
Logging this error in the main logger can be a good solution?

Reply:
If we fail to convert it, we should enforce that the query is not explainable though (as we cannot bind the parameters)

Reply:
Ok, now setting the query as non-explainable in `catch` block